File: 2648. Generate Fibonacci Sequence.rb
#* 2648. Generate Fibonacci Sequence
callCount = 5
#*
def fib_generator
    Fiber.new do
      a, b = 0, 1
      loop do
        Fiber.yield a
        a, b = b, a + b
      end
    end
end

gen = fib_generator
puts gen.resume  # Output: 0
puts gen.resume  # Output: 1

File: 2648. Generate Fibonacci Sequence.js
// 2648. Generate Fibonacci Sequence
callCount = 5
/**
 * @return {Generator<number>}
 */
var fibGenerator = function*() {
    let a = 0, b = 1
    while (true) {
        yield a;
        [a, b] = [b, a+b]
    }
};

/**
 * const gen = fibGenerator();
 * gen.next().value; // 0
 * gen.next().value; // 1
 */